tests/test_ethereum_payment_provider.py

```diff
@@ -0,0 +1,137 @@
+import datetime
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import timezone
+from django.test import RequestFactory
+from django.contrib.sessions.backends.db import SessionStore
+
+import pytest
+
+from pretix.base.models import (
+    Event,
+    Organizer,
+)
+
+from pretix_eth.payment import Ethereum
+
+
+ETH_ADDRESS = '0xeee0123400000000000000000000000000000000'
+DAI_ADDRESS = '0xda10123400000000000000000000000000000000'
+
+
+@pytest.fixture
+def event(transactional_db):
+    now = timezone.now()
+    presale_start_at = now + datetime.timedelta(days=2)
+    presale_end_at = now + datetime.timedelta(days=6)
+    start_at = now + datetime.timedelta(days=7)
+    end_at = now + datetime.timedelta(days=14)
+
+    organizer = Organizer.objects.create(name='Ethereum Foundation')
+    event = Event.objects.create(
+        name='Devcon',
+        organizer=organizer,
+        date_from=start_at,
+        date_to=end_at,
+        presale_start=presale_start_at,
+        presale_end=presale_end_at,
+        location='Osaka',
+    )
+    return event
+
+
+@pytest.fixture
+def provider(event):
+    provider = Ethereum(event)
```

Comment:
`Ethereum` seems like a problematic name for a class, but seems out of scope for the PR. I suppose you could import `payment` so it would be `payment.Ethereum(event)` here, if you like that better.